### PR TITLE
chore: adapt ci/cd logic

### LINKED ISSUE #534: Adapt notebooks CI/CD pipeline
- We currently issue an automatically created PR to update the renku requirements after each successful PR merge. This should only happen on tag. 
- The CI deployments should run against `@master` of renku instead of `@development`.

### PR DESCRIPTION
**Pull request by ableuler**
- Update renku requirements only when a new version is released.
- Run ci deployements / acceptance tests against the main branch in renku.

closes #534.

### CODE REVIEW

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: CD
+
+on: [release]
+
+jobs:
+  publish-chart:
+    runs-on: ubuntu-latest
+    needs: test-chart
```
Comment (rokroskar) [inline .github/workflows/cd.yml:8]: you're missing this step - copy it over?

File: .github/workflows/cd.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      with:
+        fetch-depth: 0
+    - name: set up environment variables
+      run: |
+        echo "GIT_USER=Renku Bot" >> $GITHUB_ENV
+        echo "GIT_EMAIL=renku@datascience.ch" >> $GITHUB_ENV
+    - name: Push chart and images
+      uses: SwissDataScienceCenter/renku/actions/publish-chart@master
+      env:
+        GITHUB_TOKEN: ${{ secrets.DEPLOY_TOKEN }}
```
Comment (rokroskar) [inline .github/workflows/cd.yml:21]: should we make this an org-level secret?

File: .github/workflows/cd.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+        DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
+    - name: Wait for chart to get published
+      run: sleep 120
+    - name: Update component version
+      uses: SwissDataScienceCenter/renku/actions/update-component-version@master
```
Comment (rokroskar) [inline .github/workflows/cd.yml:27]: The PR that is triggered by this action will need to be changed back to merge to `master`

**Comment by ableuler** [comment]
oh wow, now you were too quick @rokroskar, sorry. Quickly changed the logic here after realising that the last step didn't build the chart and thus relies on the previous ones.

File: .github/workflows/cd.yml (hunk 3 above)
Comment (ableuler) [reply .github/workflows/cd.yml]: Yep - the corresponding issue is here: https://github.com/swissdatasciencecenter/renku/issues/1914

File: .github/workflows/cd.yml (hunk 1 above)
Comment (ableuler) [reply .github/workflows/cd.yml]: Yep, noticed this too and decided to remain with "push" but filter on `tags` only.

File: .github/workflows/cd.yml (hunk 2 above)
Comment (ableuler) [reply .github/workflows/cd.yml]: yep, seems like a good idea.

**Review by rokroskar — APPROVED** [review]

**Pull request merged**